iscservice/isc.c

```diff
@@ -156,13 +161,19 @@ static int isc_comp_launch(struct m0_isc_comp_req *comp_req, struct m0_fom *fom,
 	}
 
 	if (M0_FI_ENABLED("oom")) {
-		m0_buf_free(&comp_req->icr_result);
 		rc = M0_ERR(-ENOMEM);
+		goto err;
```

Comment:
Please note, if we have this change, the next line "m0_fom_phase_set(fom, next_phase);" will not be executed.

Is this right?

Reply:
I think this is handled in isc_comp_prepare():
Line 234 ~ 236.

Reply:
Yes, there is no point to go to the next phase if we have the error. We go immediately to the failure phase in this case.

Reply:
OK